core/templates/cowdata.h

```diff
@@ -209,30 +220,12 @@ class CowData {
 	template <bool p_ensure_zero = false>
 	Error resize(Size p_size);
 
-	_FORCE_INLINE_ void remove_at(Size p_index) {
-		ERR_FAIL_INDEX(p_index, size());
-		T *p = ptrw();
-		Size len = size();
-		for (Size i = p_index; i < len - 1; i++) {
-			p[i] = std::move(p[i + 1]);
-		}
-
-		resize(len - 1);
-	}
+	[[nodiscard]] Error reserve(USize p_min_capacity);
```

Comment:
Is `[[nodiscard]` really needed? For most use cases, I think this just enforces unnecessary error handling.

Reply:
A failed `reserve` allocation is almost certainly going to lead to undefined behavior afterwards (e.g. from failed `push_back` calls), so it would be good if callers need to consciously handle this.

Other, similar APIs such as `resize` currently don't force callers to handle this case, which I think is a mistake.  Of course, this is somewhat of a departure from our current API design, and it may turn out to be annoying for other developers. I still think it's the right call, but I'm open to discussing or changing this!

Reply:
I also agree with @DeeJayLSP and wouldn't add the `[[nodiscard]]`.

Effectively if this happens, if this doesn't crash, something else will .. I think as an engine Godot has positively taken the decision to not handle out of memory conditions (in most cases), and so forcing call sites to add error handling code goes against this.

Reply:
Ok, I made it non-nodiscard.